cli/tests/pcluster/delete/test_pcluster_delete.py

```diff
@@ -0,0 +1,141 @@
+"""This module provides unit tests for the functions in the pcluster.delete module."""
+
+from collections import namedtuple
+
+import pytest
+
+import pcluster.commands as commands
+import pcluster.utils as utils
+from assertpy import assert_that
+
+FakePdeleteArgs = namedtuple("FakePdeleteArgs", "cluster_name config_file nowait keep_logs region")
+FAKE_CLUSTER_NAME = "cluster_name"
+FAKE_STACK_NAME = utils.get_stack_name(FAKE_CLUSTER_NAME)
+LOG_GROUP_TYPE = "AWS::Logs::LogGroup"
+
+
+def get_fake_pdelete_args(cluster_name="cluster_name", config_file=None, nowait=False, keep_logs=False, region=None):
+    """Get a FakePdeleteArgs instance, with None used for any parameters not specified."""
+    return FakePdeleteArgs(
+        cluster_name=cluster_name, config_file=config_file, nowait=nowait, keep_logs=keep_logs, region=region
+    )
+
+
+@pytest.mark.parametrize(
+    "keep_logs,stack_exists,warn_call_count,persist_called,delete_called",
+    [
+        (True, False, 2, False, False),
+        (False, False, 1, False, False),
+        (False, True, 0, False, True),
+        (True, True, 0, True, True),
+    ],
+)
+def test_delete(mocker, keep_logs, stack_exists, warn_call_count, persist_called, delete_called):
+    """Verify that commands.delete behaves as expected."""
+    mocker.patch("pcluster.commands.utils.stack_exists").return_value = stack_exists
+    mocker.patch("pcluster.commands._persist_cloudwatch_log_groups")
+    mocker.patch("pcluster.commands._delete_cluster")
+    mocker.patch("pcluster.commands.utils.warn")
+    mocker.patch("pcluster.commands.PclusterConfig.init_aws")
+    args = get_fake_pdelete_args(keep_logs=keep_logs)
+
+    if not stack_exists:
+        with pytest.raises(SystemExit) as sysexit:
+            commands.delete(args)
+        assert_that(sysexit.value.code).is_equal_to(0)
+    else:
+        commands.delete(args)
+
+    commands.PclusterConfig.init_aws.called_with(args.config_file)
+
+    assert_that(commands.utils.warn.call_count).is_equal_to(warn_call_count)
+    if warn_call_count > 0:
+        commands.utils.warn.assert_called_with("Cluster {0} has already been deleted.".format(args.cluster_name))
+
+    assert_that(commands._persist_cloudwatch_log_groups.called).is_equal_to(persist_called)
+    if persist_called:
+        commands._persist_cloudwatch_log_groups.assert_called_with(args.cluster_name)
+
+    assert_that(commands._delete_cluster.called).is_equal_to(delete_called)
+    if delete_called:
+        commands._delete_cluster.assert_called_with(args.cluster_name, args.nowait)
+
+
+@pytest.mark.parametrize(
+    "stacks,template",
+    [
+        ([], {},),
+        (
+            [{"StackName": FAKE_STACK_NAME}, {"StackName": "substack_one"}],
+            {"Resources": {"key": {"DeletionPolicy": "Retain"}}},
+        ),
+        ([{"StackName": FAKE_STACK_NAME}], {"Resources": {"key": {"DeletionPolicy": "Don't Retain"}}},),
```

Comment:
minor: Don't Retain --> Delete? to use a real value

Reply:
Same as the other responses.